openedx_learning/core/itemstore/models.py

```diff
@@ -0,0 +1,238 @@
+                name="lcviv_uniq_lcv_iv",
+            ),
+
+            # An Item should have at most one version of itself published as
+            # part of any given LearningContextVersion. Having multiple
+            # ItemVersions from the same Item in a given LearningContextVersion
+            # would cause the identifiers to collide, which could cause buggy
+            # behavior without much benefit.
+            # 
+            # Ideally, we could enforce this with a constraint condition that
+            # queried item_version.item, but MySQL does not support this. So we
+            # waste a little extra space to help enforce data integrity by
+            # adding a foreign key to the Item directly in this model, and then
+            # checking the uniqueness of (LearningContextVersion, Item).
+            models.UniqueConstraint(
+                fields=["learning_context_version", "item"],
+                name="lcviv_uniq_lcv_item"
+            )
+        ]
+
+
+class Content(models.Model):
+    """
+    This is the most basic piece of raw content data, with no version metadata.
+
+    Content stores data in an immutable Binary BLOB `data` field. This data is
+    not auto-normalized in any way, meaning that pieces of content that are
+    semantically equivalent (e.g. differently spaced/sorted JSON) will result in
+    new entries. This model is intentionally ignorant of what these things mean,
+    because it expects supplemental data models to build on top of it.
+
+    Two Content instances _can_ have the same hash_digest if they are of
+    different MIME types. For instance, an empty text file and an empty SRT file
+    with both hash the same way, but be considered different entities.
+
+    The other fields on Content are for data that is intrinsic to the file data
+    itself (e.g. the size). Any smart parsing of the contents into more
+    structured metadata should happen in other models that hang off of ItemInfo.
+ 
+    Content models are not versioned in any way. The concept of versioning
+    exists at a higher level.
+
+    Since this model uses a BinaryField to hold its data, we have to be careful
+    about scalability issues. For instance, video files should not be stored
+    here directly. There is a 10 MB limit set for the moment, to accomodate
+    things like PDF files and images, but the itention is for the vast majority
+    of rows to be much smaller than that.
+    """
+    # Cap item size at 10 MB for now.
+    MAX_SIZE = 10_000_000
+
+    learning_context = models.ForeignKey(LearningContext, on_delete=models.CASCADE)
+    hash_digest = hash_field()
+
+    # Per RFC 4288, MIME type and sub-type may each be 127 chars. Add one more
+    # char for the '/' in the middle, and we're at 255.
+    #
+    # TODO: maybe we should split it up into two fields instead of doing
+    #       substring searches? 
+    mime_type = models.CharField(max_length=255, blank=False, null=False)
```

Comment:
> maybe we should split it up into two fields instead of doing substring searches? 

Sounds like a win to me. Any downsides?

Another reason to split: [MIME types can include a `parameter` field](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types#structure_of_a_mime_type), eg `text/plain;charset=utf8`. If you don't want to support the parameter field, then splitting into two fields would make that more explicit.

Reply:
I guess I'm just used to thinking them as a combined thing because I can't really see querying one in isolation from the other. But it is more normalized and it's easy to add, so 🤷 I'll add it.